auth/standard_role_manager.cc

```diff
@@ -480,6 +482,44 @@ standard_role_manager::modify_membership(
     co_await when_all_succeed(modify_roles, modify_role_members).discard_result();
 }
 
+future<>
+standard_role_manager::modify_membership(
+        std::string_view grantee_name,
+        std::string_view role_name,
+        membership_change ch,
+        ::service::mutations_collector& mc) {
+    if (legacy_mode(_qp)) {
+        co_return co_await legacy_modify_membership(grantee_name, role_name, ch);
+    }
+
+    const auto modify_roles = format(
+            "UPDATE {}.{} SET member_of = member_of {} ? WHERE {} = ?",
+            get_auth_ks_name(_qp),
+            meta::roles_table::name,
+            (ch == membership_change::add ? '+' : '-'),
+            meta::roles_table::role_col_name);
+    co_await collect_mutations(_qp, mc, modify_roles,
+            {role_set{sstring(role_name)}, sstring(grantee_name)});
+
+    sstring modify_role_members;
+    switch (ch) {
+    case membership_change::add:
+        modify_role_members = format("INSERT INTO {}.{} (role, member) VALUES (?, ?)",
+                get_auth_ks_name(_qp),
+                meta::role_members_table::name);
+        break;
+    case membership_change::remove:
+        modify_role_members = format("DELETE FROM {}.{} WHERE role = ? AND member = ?",
+                get_auth_ks_name(_qp),
+                meta::role_members_table::name);
+        break;
+    default:
+        on_internal_error(log, format("unknown membership_change value: {}", int(ch)));
+    }
+    co_await collect_mutations(_qp, mc, modify_role_members,
```

Comment:
Having dynamically parsed statements rather than using prepared statements is against our own guidelines. Are they cached further down the line?

Reply:
yes, in query_processor::get_mutations_internal, `auto stmt = prepare_internal(query_string);`

Reply:
ok. It's not beautiful but it's not practical to try for better now.